docs/docs/reference/built-in-components/gatsby-script.md

```diff
@@ -0,0 +1,196 @@
+
+- Via React's special [`dangerouslySetInnerHTML`](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml) property
+- Via a template literal
+
+Here's a look at both:
+
+```jsx
+<Script id="first-unique-id" dangerouslySetInnerHTML={{ __html: `alert('Hello world')` }} />
+<Script id="second-unique-id">{`alert('Hello world')`}</Script>
+```
+
+Functionally, both of these ways of defining inline scripts are equivalent.
+
+## Strategies
+
+You can declare a loading strategy by passing a `strategy` property. These are the available loading strategies:
+
+- `post-hydrate` - Loads after the page has hydrated
+- `idle` - Loads after the page has become idle
+- `off-main-thread` - Loads off the main thread in a web worker via [Partytown](https://partytown.builder.io)
+
+The best strategy to use depends on the functionality of the script you would like to include.
+
+A general rule of thumb is if your script can afford to be loaded later (as many analytics scripts can be), then start with `off-main-thread` or `idle` and move to earlier loading strategies if necessary.
+
+Here's how you can define these strategies in the `<Script>` component:
+
+```jsx
+<Script src="https://my-example-script" strategy="post-hydrate" />
+<Script src="https://my-example-script" strategy="idle" />
+<Script src="https://my-example-script" strategy="off-main-thread" />
+```
+
+Additionally, Gatsby exports a `ScriptStrategy` enum that you can use in TSX files if you prefer:
+
+```tsx
+import { Script, ScriptStrategy } from "gatsby"
+
+<Script src="https://my-example-script" strategy={ScriptStrategy.postHydrate} />
+<Script src="https://my-example-script" strategy={ScriptStrategy.idle} />
+<Script src="https://my-example-script" strategy={ScriptStrategy.offMainThread} />
+```
+
+### Off main thread strategy
+
+The `off-main-thread` strategy leverages [Partytown](https://partytown.builder.io) under the hood, so it may require further configuration. See the [Partytown configuration](https://partytown.builder.io/configuration) documentation for complete details.
+
+Here is an example configuring the `<Script>` component with [Google Analytics](https://analytics.google.com/analytics/web/):
+
+```tsx
+import { Script, ScriptStrategy } from "gatsby"
+
+const GTM = `G-XXXXXXXXXX`; // Your Google Analytics 4 identifier
+const GTAG = `https://www.googletagmanager.com/gtag/js?id=${GTM}`
+
+<Script src={GTAG} strategy={ScriptStrategy.offMainThread} forward={[`gtag`]} debug={true} />
+<Script id="gtag-config" strategy={ScriptStrategy.offMainThread}>
+{`
+  window.dataLayer = window.dataLayer || [];
+  window.gtag = function gtag() { window.dataLayer.push(arguments); }
+  gtag('js', new Date());
+  gtag('config', ${GTM}, { send_page_view: false })
+`}
+</Script>
+```
+
+See the [Gatsby blog post on the topic](https://www.gatsbyjs.com/blog/how-to-add-google-analytics-gtag-to-gatsby-using-partytown/) for more context.
+
+#### Forward collection
```

Comment:
Add note of limitation that if off-main-thread scripts are added during CSR then forwards will not be collected